File: data_structures/trees/python/traversal.py
def traverse_in_order(root):
    if root != None:
        traverse_in_order(root.left)
        print(root.value, end=" ")
        traverse_in_order(root.right)
    
def traverse_pre_order(root):
    if root != None:
        print(root.value, end=" ")
        traverse_pre_order(root.left)
        traverse_pre_order(root.right)

File: data_structures/trees/golang/traversal.go
package main

import "fmt"

// TraverseInOrder method to be exported
func TraverseInOrder(root *BinarySearchTreeNode) {
	if root == nil {
		return
	}
	TraverseInOrder(root.left)
	fmt.Printf("%d ", root.value)
	TraverseInOrder(root.right)
}
